src/main/java/io/spaship/operator/service/k8s/Operator.java

```diff
@@ -37,414 +36,413 @@
+    private final String deDebugNs;
+
+    public Operator(@Named("default") OpenShiftClient ocClient,
+                    EventManager eventManager, @Named("deNamespace") String ns) {
+        this.ocClient = ocClient;
+        this.eventManager = eventManager;
+        domain = ConfigProvider.getConfig().getValue("operator.domain.name", String.class);
+        routerDomain = ConfigProvider.getConfig().getValue("operator.router.domain.name", String.class);
+        routerShard = ConfigProvider.getConfig().getValue("operator.router.shard.type", String.class);
+        appInstance = setAppInstanceValue();
+        this.deDebugNs = ns;
     }
-    return propertyValue;
-  }
-
-  public OperationResponse createOrUpdateEnvironment(Environment environment) {
-
-    propertyValidation();
-
-    ReUsableItems.enforceOpsLocking(new Pair<>(environment.getIdentification(), environment.getTraceID()));
-
-    boolean envExists = environmentExists(environment);
-    LOG.debug("envExists is {}", envExists);
-    if (!envExists)
-      createNewEnvironment(environment);
-    String sideCarSvcUrl = environmentSidecarUrl(environment);
-    environment.setOperationPerformed(true);
-
-    ReUsableItems.releaseLock(environment.getIdentification());
-    LOG.debug("\n");
-    return OperationResponse.builder().environment(environment).sideCarServiceUrl(sideCarSvcUrl)
-        .originatedFrom(this.getClass().toString()).status(envExists ? 2 : 1).build();
-  }
-
-  private void propertyValidation() {
-    propertyValidation(domain, "operator.domain.name");
-    propertyValidation(routerDomain, "operator.router.domain.name");
-    propertyValidation(routerShard, "operator.router.shard.type");
-  }
-
-  private void propertyValidation(String property, String propertyID) {
-    if (Objects.isNull(property)) {
-      LOG.error("the property value of {} is missing", propertyID);
-      throw new ResourceNotFoundException("property " + propertyID + " is not set");
+
+    private String setAppInstanceValue() {
+        var propertyValue = ConfigProvider.getConfig().getValue("app.instance", String.class);
+        if (Objects.isNull(propertyValue) || propertyValue.isEmpty() || propertyValue.isBlank()) {
+            LOG.warn("property app.instance is not set, going with the default value");
+            propertyValue = "default";
+        }
+        return propertyValue;
+    }
+
+    public OperationResponse createOrUpdateEnvironment(Environment environment) {
+
+        propertyValidation();
+
+        ReUsableItems.enforceOpsLocking(new Pair<>(environment.getIdentification(), environment.getTraceID()));
+
+        boolean envExists = environmentExists(environment);
+        LOG.debug("envExists is {}", envExists);
+        if (!envExists)
+            createNewEnvironment(environment);
+        String sideCarSvcUrl = environmentSidecarUrl(environment);
+        // todo mutability is not good, we should not change the state of the object
```

Comment:
Optional : As you are following `// TODO:` format in the entire code, you can format `// todo` with `// TODO:`

Reply:
ack